static/js/chart/ranking_table.tsx

```diff
@@ -0,0 +1,145 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Component for rendering a single ranking list.
+ */
+
+import React from "react";
+
+import { formatNumber, LocalizedLink } from "../i18n/i18n";
+import { formatString } from "../topic_page/string_utils";
+
+const NUM_FRACTION_DIGITS = 2;
+
+export interface Point {
+  dcid: string;
+  label: string;
+  value: number;
+  /**
+   * If not provided, the component will calculate the rank based on the order of the input points.
+   */
+  rank?: number;
```

Comment:
I tried to create a component that can be used by both the topic page and the place explorer. The API used by the topic page does not return rank. So I make rank optional.
For the redirect link, what I'm thinking about is if there is a situation where the label is not a place and we do not need the link for that label?  
I want to make the ranking table more general. But if it doesn't make sense, I can change it.

Reply:
I think for a ranking data point, it would make sense for there to be a rank field. For the ranking in topic pages, each data point is actually generated from the fetched data, so could add an additional rank field to the object: https://github.com/datacommonsorg/website/blob/master/static/js/topic_page/ranking_tile.tsx#L151

I think the redirect link is not necessary because currently both topic page and place page is doing the same redirect. I think when/if the time comes where we want to re-use this component to redirect to a different link, the component can just take in an additional prop such as `getRedirect: (point: Point) => string`

Reply:
Done